tests/describes.js

```diff
@@ -0,0 +1,91 @@
+// -----------------------------------------------------------------------------------------
+// #region Imports
+// -----------------------------------------------------------------------------------------
+
+const {
+    ERROR_OUTPUT_STRING,
+    HELP_DESCRIPTION,
+    HELP_OPTIONS,
+} = require("../_modules/constants");
+const testUtils = require("./test-utils");
+
+// #endregion Imports
+
+// -----------------------------------------------------------------------------------------
+// #region Functions
+// -----------------------------------------------------------------------------------------
+
+/**
+ * Describe/test block for testing functionality of a command triggered by one or more options
+ * from an array.
+ *
+ * @param {any[]} options Array of options to generate `describe` blocks for. Usually a string.
+ * @param {(any) => void} fn Function to be run nested under the generated `describe` block. Can be
+ * a test case or another `describe` block.
+ */
+const _whenGivenOptions = (options, fn) => {
```

Comment:
This is minor, but due to GWT (given, when, then) best to rename this. I believe this might just be `_givenOptions`. Being this is setup. I'm definitely not perfect around GWT wording. More often then not I use 'when' where I should probably just be using 'given'.

---

GWT is composed of three concepts or steps:
Given: A precondition for the test (context)
When: The action to perform (action)
Then: A result to expect (check)

Reply:
Ah, fair enough! I had always thought of 'when' as the precondition for the test.